src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -205,4 +213,151 @@ public boolean isSelected() {
+
+        JSONObject json = req.getSubmittedForm();
+        String goal = json.get("goalType").toString();
+
+        if("download".equals(goal)){
+            doDownloadBundles(req,rsp);
+        }
+        else{
+            doDeleteBundles(req,rsp);
+        }
+    }
+
+    @Restricted(NoExternalUse.class)
+    public void doDownloadBundles(StaplerRequest req, StaplerResponse rsp) throws ServletException, IOException {
+        if (!"POST".equals(req.getMethod())) {
+            rsp.sendRedirect2(".");
+            return;
+        }
+        JSONObject json = req.getSubmittedForm();
+        String goal = json.get("goalType").toString();
+
+        if (!json.has("components")) {
+            rsp.sendError(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        List<Integer> indexList = getSelectedIndices(req);
+        rsp.setContentType("application/zip");
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd_HH.mm.ss");
+        dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
+        rsp.addHeader("Content-Disposition", "inline; filename="+BUNDLE_COLLECTION_NAME+"_"+dateFormat.format(new Date())+".zip;");
+        final ServletOutputStream servletOutputStream = rsp.getOutputStream();
+        BundleBrowser bundleBrowser = BundleBrowser.getBundleBrowser();
+        try {
+            SupportPlugin.setRequesterAuthentication(Jenkins.getAuthentication());
+            try {
+                SecurityContext old = ACL.impersonate(ACL.SYSTEM);
+                try {
+                    SupportPlugin.writeBundleCollection(servletOutputStream,bundleBrowser.getSelectedFiles(indexList));
+                } catch (IOException e) {
+                    logger.log(Level.FINE, e.getMessage(), e);
+                } finally {
+                    SecurityContextHolder.setContext(old);
+                }
+            } finally {
+                SupportPlugin.clearRequesterAuthentication();
+            }
+        } finally {
+            logger.fine("Response completed");
+        }
+    }
+
+    @Restricted(NoExternalUse.class)
+    public void doDeleteBundles(StaplerRequest req, StaplerResponse rsp) throws ServletException, IOException {
+        if (!"POST".equals(req.getMethod())) {
+            rsp.sendRedirect2(".");
+            return;
+        }
+        JSONObject json = req.getSubmittedForm();
+        if (!json.has("components")) {
+            rsp.sendError(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        List<Integer> indexList = getSelectedIndices(req);
+        BundleBrowser bundleBrowser = BundleBrowser.getBundleBrowser();
+        bundleBrowser.deleteBundle(indexList);
+        PrintWriter out = rsp.getWriter();
+        out.println("</script>");
+        rsp.forwardToPreviousPage(req);
+
+        rsp.setContentType("text/html");
+        out.println("<script type=\"text/javascript\">");
+        out.println("alert('Bundle(s) Deleted!');");
```

Comment:
I would instead create a new IOException which the message says that the bundle was already deleted. Let the exception bubble up as Jenkins knows how to handle these exceptions.